### ISSUE TITLE
flow: Cannot call getGraphQLParams(...).then with function bound to onFulfill

### ISSUE THREAD

**Issue by langpavel** [graphql/express-graphql#448]
`$ flow check --show-all-branches`
`Error ┈ node_modules/express-graphql/dist/index.js.flow:158:13`

Cannot call `getGraphQLParams(...).then` with function bound to `onFulfill` because in the return value:
 - Either `OptionsData` `[1]` is incompatible with `Promise` `[2]`.
 - Or cannot call `validationRules.concat` because in type argument `Item`:
    - Either cannot call `validate` with `validationRules` bound to `rules` because `mixed` `[3]` is incompatible with `ValidationRule` `[4]` in array element.
    - Or cannot call `validate` with `validationRules` bound to `rules` because `array type` `[5]` is incompatible with `ValidationRule` `[4]` in array element.

```
        node_modules/express-graphql/dist/index.js.flow
    [1]  43│     ) => OptionsResult)
           :
 [5][3]  78│   validationRules?: ?Array<mixed>,
           :
        151│
        152│     // Parse the Request to get GraphQL request parameters.
        153│     return getGraphQLParams(request)
        154│       .then(graphQLParams => {
        155│         params = graphQLParams;
        156│         // Then, resolve the Options to get OptionsData.
        157│         return typeof options === 'function'
        158│           ? options(request, response, params) // 🔥
        159│           : options;
        160│       })
        161│       .then(optionsData => {
        162│         // Assert that optionsData is in fact an Object.
        163│         if (!optionsData || typeof optionsData !== 'object') {

        /tmp/flow/flowlib_27f728ff/core.js
    [2] 595│       onFulfill: (value: R) => Promise<U> | U,

        node_modules/graphql/validation/validate.js.flow
    [4]  40│   rules?: $ReadOnlyArray<ValidationRule> = specifiedRules,
```

* flow 0.80.0
* express-graphql 0.6.12
* graphql 14.0.0
* everything at latest version (2018-09-04 02:40 UTC)

**Comment by IvanGoncharov** [comment]
@langpavel Thanks for reporting 👍 
Fixed on the master by https://github.com/graphql/express-graphql/commit/5c04d1838020f691cc3dc9b098635593a75e7a1d
I will try to release a new version in the next few days.

**Comment by langpavel** [comment]
Thanks!

**Comment by CoericK** [comment]
@IvanGoncharov Sorry to ask about this, but was this released?

**Comment by IvanGoncharov** [comment]
@CoericK It was released yesterday as `0.7.1` 📦

**Comment by CoericK** [comment]
Thanks @IvanGoncharov!
